src/Elastic.Transport/Diagnostics/OpenTelemetry/OpenTelemetryData.cs

```diff
@@ -1,23 +0,0 @@
-// Licensed to Elasticsearch B.V under one or more agreements.
-// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
-// See the LICENSE file in the project root for more information
-
-using System.Collections.Generic;
-
-namespace Elastic.Transport.Diagnostics;
-
-/// <summary>
-/// Allows consumers to pass specific values for OpenTelemetry instrumentation for each request.
-/// </summary>
-public readonly struct OpenTelemetryData
-{
-	/// <summary>
-	/// The name to use for spans relating to a request.
-	/// </summary>
-	public string? SpanName { get; init; }
```

Comment:
@stevejgordon How am I supposed to set a custom `OperationName` with the new design? The property has no setter. Should I use `DisplayName` to modify it?

Reply:
Update: `DisplayName` does not update the original `OperationName` which causes a test to fail in the client library. We have to re-introduce a way to provide the `OperationName`.

A potential solution would be to bring back `OpenTelemetryData`, but modify it to, besides `OperationName`, have the current `Action<>` as a property instead of the previous "attributes" dictionary.